tests/contrib/funsor/test_tmc.py

```diff
@@ -0,0 +1,142 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import math
+
+import pytest
+import torch
+from torch.autograd import grad
+from torch.distributions import constraints
+
+from tests.common import assert_equal
+
+# put all funsor-related imports here, so test collection works without funsor
+try:
+    import funsor
+    import pyro.contrib.funsor
+    from pyroapi import distributions as dist
+    from pyroapi import infer, pyro, pyro_backend
+    funsor.set_backend("torch")
+except ImportError:
+    pytestmark = pytest.mark.skip(reason="funsor is not installed")
+
+torch.set_default_dtype(torch.float32)
```

Comment:
I suspect this is the source of the mysterious nonlocal CI failures, but I may have to make a fix upstream in Funsor to remove it.

Reply:
Why is this required? We should be testing everything with double precision on CI, as specified at https://github.com/pyro-ppl/pyro/blob/dev/tests/conftest.py#L13 If you really want local tests to use low precision, you should add that as a fixture with a context manager or sth

Reply:
It should not be required, and was a workaround for a Funsor issue that should be fixed before this merges.